### PR TITLE
Initial Python setup

### PR DESCRIPTION
**Pull request by happz**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+recursive = true
+hang-closing = true
+aggressive = 2
+
+[tool.pep8]
+max-line-length = "99"
+
+[tool.ruff]
+line-length = 99
+src = ["tmt", "tests"]
```
Comment (psss) [inline pyproject.toml:125]: A `tmt` leftover?

**Review by psss — CHANGES_REQUESTED** [review]
Thanks for preparing the skeleton! Looks good, just one typo.

**Review by thrix — CHANGES_REQUESTED** [review]

**Comment by thrix** [comment]
@happz sorry added a commit by an accident, rather reverted then rebasing ... feel free to remove

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+[project]
+name = "newa"
+dynamic = ["version"]
+description = "newa"
+readme = "README.md"
+# TODO: provide correct values
+# license = "MIT"
+# license-files = { paths = ["LICENSE"] }
+# requires-python = ">=3.9"
```
Comment (happz) [inline pyproject.toml:13]: @kkaarreell what Python versions do you expect to use? What would be the minimal one? 3.9 seems like a good one, it certainly works for tmt.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        pass_filenames: false
+        args: [--config-file=pyproject.toml]
+
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.1.11
+    hooks:
+      - id: ruff
+        args:
+          - '--fix'
+          - '--show-fixes'
```
[the remark this answers is not in the document]
Comment (happz) [reply .pre-commit-config.yaml:32]: Good point, added in b43fd39

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+# authors = [
+#     { name = "...", email = "..." },
+#     ]
+# maintainers = [
+#     { name = "...", email = "..." },
+#     ]
```
Comment (happz) [inline pyproject.toml:19]: @kkaarreell who would be authors/maintainers?

File: pyproject.toml (hunk 1 above)
Comment (happz) [reply pyproject.toml]: Fixed in b43fd39

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    "autopep8",
+    "ruff",
+    "mypy",
+    "pytest",
+    "pre-commit",
+    ]
+
+[tool.hatch.envs.dev.scripts]
+lint = ["autopep8 {args:.}", "ruff --fix {args:.}"]
+type = ["mypy {args:tmt}"]
```
[the remark this answers is not in the document]
Comment (happz) [reply pyproject.toml]: Fixed in b43fd39

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+name = "newa"
+dynamic = ["version"]
+description = "newa"
+readme = "README.md"
+# TODO: provide correct values
+# license = "MIT"
```
[the remark this answers is not in the document]
Comment (kkaarreell) [reply pyproject.toml:11]: Hi, I have added it to the next mtg agenda so its not just on me to decide.

File: pyproject.toml (hunk 2 above)
Comment (kkaarreell) [reply pyproject.toml:13]: I would like to preserve (eventually) the option to run tests using `tmt` which means we should use the same requirement.

File: pyproject.toml (hunk 4 above)
Comment (kkaarreell) [reply pyproject.toml:19]: @happz I think we can populate the initial set with names of current contributors (you, Miro, Patrik, me, possibly Lukas) and extend it later.

File: pyproject.toml (hunk 6 above)
[the remark this answers is not in the document]
Comment (happz) [reply pyproject.toml]: Added in d64dd81

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+dynamic = ["version"]
+description = "newa"
+readme = "README.md"
+# TODO: provide correct values
+license = "Apache-2.0"
+license-files = { paths = ["LICENSE.txt"] }
+requires-python = ">=3.9"
+authors = [
+    { name = "Karel Srot", email = "ksrot@redhat.com" },
+    { name = "Patrik Kis", email = "ksrotpkisredhat.com" },
```
Comment (thrix) [inline pyproject.toml:16]:
```suggestion
    { name = "Patrik Kis", email = "pkis@redhat.com" },
```

File: pyproject.toml [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+    { name = "Karel Srot", email = "ksrot@redhat.com" },
+    { name = "Miroslav Vadkerti", email = "mvadkert@redhat.com" },
+    { name = "Milos Prchlik", email = "mprchlik@redhat.com" },
+]
+
+maintainers = [
+    { name = "Karel Srot", email = "ksrot@redhat.com" },
+    { name = "Patrik Kis", email = "ksrotpkisredhat.com" },
```
Comment (thrix) [inline pyproject.toml:24]:
```suggestion
    { name = "Patrik Kis", email = "pkis@redhat.com" },
```

**Review by thrix — APPROVED** [review]
lgtm, 2 small typos

**Review by kkaarreell — APPROVED** [review]
Thank you guys

File: pyproject.toml [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+description = "newa"
+readme = "README.md"
+# TODO: provide correct values
+license = "Apache-2.0"
+license-files = { paths = ["LICENSE.txt"] }
+requires-python = ">=3.9"
+authors = [
+    { name = "Karel Srot", email = "ksrot@redhat.com" },
+    { name = "Patrik Kis", email = "pkis@redhat.com" },
+    { name = "Karel Srot", email = "ksrot@redhat.com" },
```
Comment (psss) [inline pyproject.toml:17]: Looks like a duplicate of the first line? Similar below.

Comment (happz) [reply pyproject.toml]: Fixed in a6ec587

**Review by psss — APPROVED** [review]

**Pull request merged**